redfish/internal/usecase/wsman_repo.go

```diff
@@ -37,27 +74,390 @@ var (
+	for i, item := range items {
+		if i >= maxArrayItems {
+			break
+		}
+
+		if itemMap, ok := item.(map[string]interface{}); ok {
+			if value := f.extractFromSingleItem(itemMap, propertyName); value != nil {
+				return value
+			}
+		}
+	}
+
+	return nil
+}
+
+// extractFromPullResponse extracts property from PullResponse structure.
+func (f *CIMExtractorFramework) extractFromPullResponse(responseMap map[string]interface{}, propertyName string) interface{} {
+	if pullResponse, ok := responseMap["PullResponse"].(map[string]interface{}); ok {
+		if items, ok := pullResponse["Items"].([]interface{}); ok {
+			return f.processItemsArray(items, propertyName)
+		}
+	}
+
+	return nil
+}
+
+// extractFromDirectItems extracts property from direct Items array.
+func (f *CIMExtractorFramework) extractFromDirectItems(responseMap map[string]interface{}, propertyName string) interface{} {
+	if items, ok := responseMap["Items"].([]interface{}); ok {
+		return f.processItemsArray(items, propertyName)
+	}
+
+	return nil
+}
+
+// extractFromNestedBody extracts property from Body -> PullResponse -> Items structure.
+func (f *CIMExtractorFramework) extractFromNestedBody(responseMap map[string]interface{}, propertyName string) interface{} {
+	if body, ok := responseMap["Body"].(map[string]interface{}); ok {
+		if pullResponse, ok := body["PullResponse"].(map[string]interface{}); ok {
+			if items, ok := pullResponse["Items"].([]interface{}); ok {
+				return f.processItemsArray(items, propertyName)
+			}
+		}
+	}
+
+	return nil
+}
+
+// extractFromMap extracts property from map or nested map structures.
+func (f *CIMExtractorFramework) extractFromMap(response interface{}, config CIMPropertyConfig) interface{} {
+	if response == nil {
+		return nil
+	}
+
+	// Handle array response directly
+	if itemsArray, ok := response.([]interface{}); ok {
+		return f.processItemsArray(itemsArray, config.CIMProperty)
+	}
+
+	return f.extractFromMapResponse(response, config.CIMProperty)
+}
+
+// extractFromMapResponse handles map-based responses with reduced complexity.
+func (f *CIMExtractorFramework) extractFromMapResponse(response interface{}, propertyName string) interface{} {
+	responseMap, ok := response.(map[string]interface{})
+	if !ok {
+		return nil
+	}
+
+	// Define extraction methods in order of preference
+	extractionMethods := []func(map[string]interface{}, string) interface{}{
+		f.extractFromPullResponse,
+		f.extractFromDirectItems,
+		f.extractFromNestedBody,
+		f.extractFromSingleItem,
+	}
+
+	// Try each extraction method
+	for _, method := range extractionMethods {
+		if value := method(responseMap, propertyName); value != nil {
+			return value
+		}
+	}
+
+	return nil
+}
+
+// ExtractMultipleProperties extracts multiple properties in a single call for efficiency.
+func (f *CIMExtractorFramework) ExtractMultipleProperties(ctx context.Context, systemID string, configs []CIMPropertyConfig) map[string]interface{} {
```

Comment:
This function is not used in any other module other than here. can you make this a private function i.e rename ExtractMultiplePropertiesto extractMultipleProperties